File: app/src/main/java/com/mouredev/weeklychallenge2022/Challenge16.java
/*
 * Reto #16
 * EN MAYÚSCULA
 * Fecha publicación enunciado: 18/04/22
 * Fecha publicación resolución: 25/04/22
 * Dificultad: FÁCIL
 *
 * Enunciado: Crea una función que reciba un String de cualquier tipo y se encargue de
 * poner en mayúscula la primera letra de cada palabra.
 * - No se pueden utilizar operaciones del lenguaje que lo resuelvan directamente.
 *
 * Información adicional:
 * - Usa el canal de nuestro discord (https://mouredev.com/discord) "🔁reto-semanal" para preguntas, dudas o prestar ayuda a la comunidad.
 * - Puedes hacer un Fork del repo y una Pull Request al repo original para que veamos tu solución aportada.
 * - Revisaré el ejercicio en directo desde Twitch el lunes siguiente al de su publicación.
 * - Subiré una posible solución al ejercicio el lunes siguiente al de su publicación.
 *
 */

/**
 * Challenge16
 */
public class Challenge16 {

    public static void main(String[] args) {
        wordsCapital("hola que tal, com va tot.");
    }

    private static void wordsCapital(String phrase) {
        String words[] = phrase.split(" ");
        String result = "";

        for (String word : words) {
            result += word.substring(0, 1).toUpperCase();
            result += word.substring(1) + " ";
        }

        System.out.println(result);
    }
}

File: app/src/main/java/com/mouredev/weeklychallenge2022/Challenge16.kt
package com.mouredev.weeklychallenge2022

import java.util.*

/*
 * Reto #16
 * EN MAYÚSCULA
 * Fecha publicación enunciado: 18/04/22
 * Fecha publicación resolución: 25/04/22
 * Dificultad: FÁCIL
 *
 * Enunciado: Crea una función que reciba un String de cualquier tipo y se encargue de
 * poner en mayúscula la primera letra de cada palabra.
 * - No se pueden utilizar operaciones del lenguaje que lo resuelvan directamente.
 *
 * Información adicional:
 * - Usa el canal de nuestro discord (https://mouredev.com/discord) "🔁reto-semanal" para preguntas, dudas o prestar ayuda a la comunidad.
 * - Puedes hacer un Fork del repo y una Pull Request al repo original para que veamos tu solución aportada.
 * - Revisaré el ejercicio en directo desde Twitch el lunes siguiente al de su publicación.
 * - Subiré una posible solución al ejercicio el lunes siguiente al de su publicación.
 *
 */

fun main() {
    println(mayusculas("hola mundo"))
    println(mayusculas("Como estais todos hoy"))
}

private fun mayusculas(texto: String) : String{
    val textoArray = texto.split(" ")
    val textoFinal = textoArray.map { palabra ->
        palabra.replaceFirstChar {
            if (it.isLowerCase())
                it.titlecase(Locale.getDefault())
            else
                it.toString()
        }
    }.joinToString(" ")
    return textoFinal
}